### PR TITLE
redirect user to login page if not connected

### PR DESCRIPTION
**Pull request by SimonLab**
see #307

Most of the changes are applied on the tests file where the 401 status code is replaces by a 302 redirection to the login page

### CODE REVIEW

File: server/handlers/challenges/add-view.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-module.exports = (request, reply, source, joiErr) => {
+module.exports = function (request, reply, source, joiErr) {
   var error = helpers.errorOptions(joiErr);
   var loggedIn = request.auth.credentials;
   var permissions = helpers.getPermissions(loggedIn, 'scope', 'admin');
+  var options = {};
+
   if (loggedIn.scope !== 'primary') {
-    return reply(Boom.unauthorized('Only Primary users can create challenges'));
+    return reply(Boom.forbidden());
```
Comment (mk4111) [inline server/handlers/challenges/add-view.js]: Just out of interest, why `forbidden` over `unauthorised`?

File: server/handlers/people/add-view.js [hunk 2]
```diff
@@ -11,9 +12,14 @@ module.exports = function (request, reply, source, joiErr) {
   var error = helpers.errorOptions(joiErr);
   var loggedIn = request.auth.credentials;
   var permissions = helpers.getPermissions(loggedIn, 'scope', 'admin');
+  var getActiveOrgs = request.server.methods.pg.organisations.getActive;
   var unlinkedActiveOrgs, options;
 
-  request.server.methods.pg.organisations.getActive(function (pgErr, orgs) {
+  if (loggedIn.scope !== 'admin') {
```
Comment (mk4111) [inline server/handlers/people/add-view.js]: Can we put this before .getActive? I think then we can move them on without needing to query the DB

File: server/handlers/challenges/add-view.js (hunk 1 above)
Comment (mk4111) [inline server/handlers/challenges/add-view.js]: Just saw the distinction you've made between 401 and 403 - I understand now :)
I liiiikey

File: server/hapi-error-config.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+module.exports = {
+  statusCodes: {
+    401: { redirect: '/login' },
+    403: { message: 'Sorry you do not have the permissions to access this page'}
+  }
+};
```
Comment (mk4111) [inline server/hapi-error-config.js]: This is really cool!!!

File: server/handlers/people/add-view.js (hunk 2 above)
Comment (SimonLab) [inline server/handlers/people/add-view.js]: It is before getActiveOrgs :arrow_down:

Comment (mk4111) [inline server/handlers/people/add-view.js]: My bad - I thought it was inside!

**Pull request merged**